xbmc/pvr/dialogs/GUIDialogPVRRecordingSettings.cpp

```diff
@@ -94,6 +94,21 @@ void CGUIDialogPVRRecordingSettings::InitializeSettings()
     setting = AddList(group, SETTING_RECORDING_LIFETIME, 19083, SettingLevel::Basic, m_iLifetime, LifetimesFiller, 19083);
 }
 
+bool CGUIDialogPVRRecordingSettings::CanEditRecording(const CFileItem& item)
+{
+  if (!item.HasPVRRecordingInfoTag())
+    return false;
+
+  const std::shared_ptr<CPVRClient> client = CServiceBroker::GetPVRManager().GetClient(item.GetPVRRecordingInfoTag()->ClientID());
+
+  bool client_capabilities = (
+    client->GetClientCapabilities().SupportsRecordingsRename() ||
+    client->GetClientCapabilities().SupportsRecordingsPlayCount() ||
+    client->GetClientCapabilities().SupportsRecordingsLifetimeChange());
```

Comment:
That's not what I ment. The idea is not to call `client->GetClientCapabilities()` three times. That's waste of CPU time. And please, no varibale names with underscores, except if it is a member variable (coding guidelines).

Reply:
Please change that. Thanks.

Reply:
Do you mean something like this:

```
  CPVRClientCapabilities capabilities = client->GetClientCapabilities();

  return client && (
    capabilities.SupportsRecordingsRename() ||
    capabilities.SupportsRecordingsPlayCount() ||
    capabilities.SupportsRecordingsLifetimeChange());
```

Reply:
Almost. => `const CPVRClientCapabilities& capabilities = ...` (no need to create a copy, reference is better as it saves ressources)

Reply:
And ofc you need the check for client not being null before you access it!

Reply:
Something like this:
```
  if (client == nullptr)
    return false;
    
  const CPVRClientCapabilities& capabilities = client->GetClientCapabilities();

  return client && (
    capabilities.SupportsRecordingsRename() ||
    capabilities.SupportsRecordingsPlayCount() ||
    capabilities.SupportsRecordingsLifetimeChange());
```

Reply:
Well, let me write it for you:

```c++
if (!client)
    return false;
    
  const CPVRClientCapabilities& capabilities = client->GetClientCapabilities();

  return capabilities.SupportsRecordingsRename() ||
             capabilities.SupportsRecordingsPlayCount() ||
             capabilities.SupportsRecordingsLifetimeChange();
```

Reply:
Of course, it was obvious, I should have seen that.